From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew Gunnerson <accounts+github@chiller3.com>
Date: Sat, 15 Aug 2026 13:37:05 -0400
Subject: [PATCH] Add ACCESS_MEDIA_LOCATION permission

MediaProvider strips out location EXIF tags by default for apps without
this permission. This makes it impossible to do a lossless file copy or
file move that requires copy + delete for images.

Adding this permission allows the user to choose whether or not to
redact EXIF location, albeit with no convenient UI.
---
 AndroidManifest.xml | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index 3f783e4e1..6d9dcc53a 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -49,6 +49,8 @@
     <!-- Permissions required for reading device configs -->
     <uses-permission android:name="android.permission.READ_DEVICE_CONFIG"/>
 
+    <uses-permission android:name="android.permission.ACCESS_MEDIA_LOCATION" />
+
     <!-- Declaring the permission for apps wanting to launch DocsUI through the home screen. -->
     <permission
         android:name="com.android.documentsui.LAUNCH_FILE_MANAGER"
-- 
2.55.0

